macos/compressed_resource.ksy

```diff
@@ -0,0 +1,182 @@
+meta:
+  id: compressed_resource
+  title: Compressed Macintosh resource
+  application: Mac OS
+  license: MIT
+  ks-version: "0.9"
+  imports:
+    - /common/data_with_io
+  endian: be
+doc: |
+  Compressed Macintosh resource data,
+  as stored in resources with the "compressed" attribute.
+
+  Resource decompression is not documented by Apple.
+  It is mostly used internally in System 7,
+  some of Apple's own applications (such as ResEdit),
+  and also by some third-party applications.
+  Later versions of Classic Mac OS make less use of resource compression,
+  but still support it fully for backwards compatibility.
+  Carbon in Mac OS X no longer supports resource compression in any way.
+
+  The data of all compressed resources starts with a common header,
+  followed by the compressed data.
+  The data is decompressed using code in a `'dcmp'` resource.
+  Some decompressors used by Apple are included in the System file,
+  but applications can also include custom decompressors.
+  The header of the compressed data indicates the ID of the `'dcmp'` resource used to decompress the data,
+  along with some parameters for the decompressor.
+doc-ref:
+  - 'http://www.alysis.us/arctechnology.htm'
+  - 'http://preserve.mactech.com/articles/mactech/Vol.09/09.01/ResCompression/index.html'
+  - 'https://github.com/dgelessus/python-rsrcfork/tree/master/rsrcfork/compress'
+seq:
+  - id: header
+    type: header
+    doc: |
+      The header of the compressed data.
+  - id: compressed_data
+    size-eos: true
+    doc: |
+      The compressed resource data.
+
+      The format of this data is completely dependent on the decompressor and its parameters,
+      as specified in the header.
+      For details about the compressed data formats implemented by Apple's decompressors,
+      see the specs in the resource_compression subdirectory.
+types:
+  header:
+    seq:
+      - id: common_part
+        type: common_part
+        doc: |
+          The common part of the header.
+          Among other things,
+          this part contains the header type,
+          which determines the format of the data in the type-specific part of the header.
+      - id: type_specific_part_raw_with_io
+        type: data_with_io
+        size: common_part.len_header - common_part._sizeof
+        doc: |
+          Use `type_specific_part_raw` instead,
+          unless you need access to this field's `_io`.
+    instances:
+      type_specific_part_raw:
+        value: type_specific_part_raw_with_io.data
+        doc: |
+          The type-specific part of the header,
+          as a raw byte array.
+      type_specific_part:
+        io: type_specific_part_raw_with_io._io
+        pos: 0
+        type:
+          switch-on: common_part.header_type
+          cases:
+            8: type_specific_part_type_8
```

Comment:
Could the names be more meaningful?

Reply:
This part of the format is practically undocumented - there are no official docs, and the unofficial docs that exist are short or incomplete (which I'm trying to fix 🙂), so there aren't any good descriptive names for this. I used more descriptive names before, but it turned out that they were inaccurate (there were some additional compression schemes that I didn't know about when I made up the names). So I switched to naming them by the number in the preceding header, because that's the only part that I know is accurate 🙂 (see dgelessus/python-rsrcfork/commit/f3b3de496ed607282413fba0ba9bf80a77c695c9)